router/status.go

```diff
@@ -0,0 +1,206 @@
+package router
+
+import (
+	"net"
+	"time"
+)
+
+type Status struct {
+	Encryption      bool
+	PeerDiscovery   bool
+	Name            string
+	NickName        string
+	Port            int
+	Interface       *net.Interface
+	MACs            []MACStatus
+	Peers           []PeerStatus
+	UnicastRoutes   []UnicastRouteStatus
+	BroadcastRoutes []BroadcastRouteStatus
+	ConnectionMaker ConnectionMakerStatus
+}
+
+type ConnectionMakerStatus struct {
+	DirectPeers []string
+	Reconnects  []TargetStatus
+}
+
+type TargetStatus struct {
+	Address     string
+	Attempting  bool      `json:"Attempting,omitempty"`
+	TryAfter    time.Time `json:"TryAfter,omitempty"`
+	TryInterval string    `json:"TryInterval,omitempty"`
+	LastError   string    `json:"LastError,omitempty"`
+}
+
+type MACStatus struct {
+	Mac      string
+	Name     string
+	NickName string
+	LastSeen time.Time
+}
+
+type PeerStatus struct {
+	Name        string
+	NickName    string
+	UID         PeerUID
+	Version     uint64
+	Connections []ConnectionStatus
+}
+
+type ConnectionStatus struct {
+	Name        string
+	NickName    string
+	Address     string
+	Outbound    bool
+	Established bool
+}
+
+type UnicastRouteStatus struct {
+	Dest, Via string
+}
+
+type BroadcastRouteStatus struct {
+	Source string
+	Via    []string
+}
+
+func NewStatus(router *Router) *Status {
+	return &Status{
+		router.UsingPassword(),
+		router.PeerDiscovery,
+		router.Ourself.Name.String(),
+		router.Ourself.NickName,
+		router.Port,
+		router.Iface,
+		NewMACStatusSlice(router.Macs),
+		NewPeerStatusSlice(router.Peers),
+		NewUnicastRouteStatusSlice(router.Routes),
+		NewBroadcastRouteStatusSlice(router.Routes),
+		NewConnectionMakerStatus(router.ConnectionMaker)}
+}
+
+func NewConnectionMakerStatus(cm *ConnectionMaker) ConnectionMakerStatus {
+	// We need to Refresh first in order to clear out any 'attempting'
+	// connections from cm.targets that have been established since
+	// the last run of cm.checkStateAndAttemptConnections. These
+	// entries are harmless but do represent stale state that we do
+	// not want to report.
+	cm.Refresh()
+	resultChan := make(chan ConnectionMakerStatus, 0)
+	cm.actionChan <- func() bool {
+		status := ConnectionMakerStatus{
+			DirectPeers: []string{},
+			Reconnects:  []TargetStatus{},
+		}
```

Comment:
you are invoking the keyless constructor for (nearly) all Status structs. so why not here?